addons/medical_damage/ACE_Medical_Injuries.hpp

```diff
@@ -7,125 +7,275 @@ class ACE_Medical_Injuries {
         class Crush {
-            causes[] = {"falling", "vehiclecrash", "collision", "punch", "unknown"};
             bleeding = 0.05;
             pain = 0.8;
-            minDamage = 0.1;
             causeLimping = 1;
             causeFracture = 1;
         };
         // Slicing wounds made with a sharp instrument, leaving even edges. They may be as minimal as a paper cut or as significant as a surgical incision.
         class Cut {
-            causes[] = {"vehiclecrash", "collision", "grenade", "explosive", "shell", "backblast", "stab", "unknown"};
             bleeding = 0.01;
             pain = 0.1;
-            minDamage = 0.1;
         };
         // Also called tears, these are separating wounds that produce ragged edges. They are produced by a tremendous force against the body, either from an internal source as in childbirth, or from an external source like a punch.
         class Laceration {
-            causes[] = {"vehiclecrash", "collision", "punch"};
             bleeding = 0.05;
             pain = 0.2;
-            minDamage = 0.01;
         };
         // Also called velocity wounds, they are caused by an object entering the body at a high speed, typically a bullet or small peices of shrapnel.
         class VelocityWound {
-            causes[] = {"bullet", "grenade","explosive", "shell", "unknown"};
             bleeding = 0.2;
             pain = 0.9;
-            minDamage = 0.35;
             causeLimping = 1;
             causeFracture = 1;
         };
         // Deep, narrow wounds produced by sharp objects such as nails, knives, and broken glass.
         class PunctureWound {
-            causes[] = {"stab", "grenade"};
             bleeding = 0.05;
             pain = 0.4;
-            minDamage = 0.02;
             causeLimping = 1;
         };
     };
+
     class damageTypes {
-        // thresholds[] {{<min damage>, <max number of wounds>}, {...}}
+        // thresholds[] {{<damage>, <number of wounds>}, {...}}
+        // if damage is between two points, number is interpolated and then rounded by chance based on the decimal part
+        // e.g. a value of 2.7 has 70% chance to give 3 and 30% to give 2
+        // put damage values in descending order; uses the first value found that is below the wound damage, and the point immediately before that
         thresholds[] = {{0.1, 1}};
+
+        // if 1, wounds are only applied to the hitpoint that took the most damage. othewrise, wounds are applied to all damaged hitpoints
         selectionSpecific = 1;
+        
+        // this can also be overridden for each damage type
+        woundsHandler = QFUNC(woundsHandlerActive);
 
         class bullet {
-            // above damage, amount. Put the highest threshold to the left and lower the threshold with the elements to the right of it.
-            thresholds[] = {{0.1, 1}};
+            thresholds[] = {{0.1, 1}, {0.1, 0}};
```

Comment:
This has the effect that bullet wounds are kind of capped - even if it does 17.5 damage it'll only ever produce one wound (although it'll definitely kill you).
I had an idea to allow bullets to create multiple wounds under certain circumstances:
- only if damage>2
- divide in such a way that each wound is always >1
- write the wound weighting so that these wounds can only be avulsions

This way very large hits (like getting your arm blown off by a 50) would create a proportional _number_ of large avulsions, allowing bleeding and pain to scale up higher and making limb shots with these types of weapons more dramatic (since they can't be fatal on their own).

Reply:
I've implemented this with higher numbers (it can always be reverted):
a bullet hit with `damage>3` may create multiple wounds, they will always be over 1.5 and wounds over that threshold always result in large avulsions. With the limb armor change, however, this virtually never happens.